### DISCUSSION TITLE
Can I use bpftrace as a strace?

### DISCUSSION THREAD

**Discussion by adudak64a** [bpftrace/bpftrace#2262] [Q&A]
I am start learning bpftrace recently and I'm interested. Can I use bpftrace as a strace?

**Comment by keyolk** [comment]
Well, I think it can be used as strace but there are some shortages.
In many case bpftrace looks more flexible than strace.
But it it is hard to follow child process of target the target whereas strace can do with its `-f` flags.

**Comment by fbs** [comment]
you can to some extend but it requires some work.

The way strace works is that it pauses execution after each systemcall, after that it inspects each argument and it prints that in a human readable way.

With bpftrace (bpf in general) you can either hook in to specific systemcalls or to the generic syscall handler but you're required to do all the syscall argument magic yourself. For some its easy, e.g. opensnoop.bt but in other cases this can become really complex. If you want to go down that road bpftrace probably isnt the best tool as you need a lot of 'post processing'.

**Comment by adudak64a** [comment]
I understood. Thank all
